File: CAEVSYNC.Common/Models/SyncRuleModel.cs
namespace CAEVSYNC.Common.Models;

public class SyncRuleModel
{
    public int Id { get; set; }
    
    public string Title { get; set; }
    
    public string OriginCalendarId { get; set; }
    
    public string TargetCalendarId { get; set; }
    
    public string OriginCalendarTitle { get; set; }
    
    public string TargetCalendarTitle { get; set; }
    
    public List<EventTransformationStepModel> EventTransformationSteps { get; set; } = new();
}

File: CAEVSYNC.Web/clientapp/src/models/SyncRuleModel.ts
import { EventTransformationStepModel } from "./EventTransformationStepModel";

export interface SyncRuleModel {
    id: number;
    title: string;
    originCalendarId: string;
    targetCalendarId: string;
    originCalendarTitle: string;
    targetCalendarTitle: string;
    eventTransformationSteps: EventTransformationStepModel[];
}